lib/current-env.js

```diff
@@ -13,20 +14,42 @@ function cpu () {
   return process.arch
 }
 
+const LDD_PATH = '/usr/bin/ldd'
+function getFamilyFromFilesystem () {
+  try {
+    const content = fs.readFileSync(LDD_PATH, 'utf-8')
+    if (content.includes('musl')) {
+      return 'musl'
+    }
+    if (content.includes('GNU C Library')) {
+      return 'glibc'
+    }
+    return null
+  } catch {
+    return undefined
+  }
+}
+
+let family
 function libc (osName) {
-  // this is to make it faster on non linux machines
   if (osName !== 'linux') {
     return undefined
   }
-  let family
-  const originalExclude = process.report.excludeNetwork
-  process.report.excludeNetwork = true
-  const report = process.report.getReport()
-  process.report.excludeNetwork = originalExclude
-  if (report.header?.glibcVersionRuntime) {
-    family = 'glibc'
-  } else if (Array.isArray(report.sharedObjects) && report.sharedObjects.some(isMusl)) {
-    family = 'musl'
+  if (family === undefined) {
+    family = getFamilyFromFilesystem()
+    if (family === undefined) {
```

Comment:
Perhaps the contents of this `if` block should be moved to a new function?

Something like `getFamilyFromReport()`